dali/operators/reader/loader/video/frames_decoder_gpu.cc

```diff
@@ -36,7 +36,11 @@ int process_video_sequence(void *user_data, CUVIDEOFORMAT *video_format) {
 int process_picture_decode(void *user_data, CUVIDPICPARAMS *picture_params) {
   FramesDecoderGpu *frames_decoder = static_cast<FramesDecoderGpu*>(user_data);
 
-  return frames_decoder->ProcessPictureDecode(user_data, picture_params);
+  if (frames_decoder->HasIndex()) {
+    return frames_decoder->ProcessPictureDecodeWithIndex(user_data, picture_params);
+  }
+
+  return frames_decoder->ProcessPictureDecodeWithoutIndex(user_data, picture_params);
```

Comment:
In this case I think that if/else is more verbose about one of the options, as it is not early exit after all.
```suggestion
  if (frames_decoder->HasIndex()) {
    return frames_decoder->ProcessPictureDecodeWithIndex(user_data, picture_params);
  } else {
    return frames_decoder->ProcessPictureDecodeWithoutIndex(user_data, picture_params);
  }
```

Reply:
Done